### PR TITLE
Parsingcsvfile ss

### PR DESCRIPTION
**Pull request by SarojKumar8101**
- Upload csv file
- Parsing CSV file and polulate model
- Save Questions in Database
- Delete uploaded csv file

Along with few bug fixes

### CODE REVIEW

File: src/Application/CbtQuestion/Command/Create/CreateCbtQuestionCommand.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace Application.CbtQuestion.Command.Create
+{
+    public class CreateCbtQuestionCommand : IRequest<int>
+    {
+        public string CsvFilePath { get; set; }    
+        public int CatalogId { get; set; }  
+       
+    }
+    public class CreateCbtQuestionCommandHandler : IRequestHandler<CreateCbtQuestionCommand, int>
```
Comment (parvez-syn) [inline src/Application/CbtQuestion/Command/Create/CreateCbtQuestionCommand.cs:17]: 💄 add space between classes and functions

File: src/Application/CbtQuestion/Command/Create/CreateCbtQuestionCommand.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+using TALMS.Application.Common.Interfaces;
+using Application.CbtQuestion.Helper;
+
+namespace Application.CbtQuestion.Command.Create
+{
+    public class CreateCbtQuestionCommand : IRequest<int>
+    {
+        public string CsvFilePath { get; set; }    
+        public int CatalogId { get; set; }  
+
```
Comment (parvez-syn) [inline src/Application/CbtQuestion/Command/Create/CreateCbtQuestionCommand.cs:15]: 💄 remove extra white space

File: src/Application/CbtQuestion/Command/Create/CreateCbtQuestionCommand.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    public class CreateCbtQuestionCommandHandler : IRequestHandler<CreateCbtQuestionCommand, int>
+    {    
+        private readonly IApplicationDbContext _context;
+        
+        private readonly ICbtQuestionParser _cbtQuestion;
+        public CreateCbtQuestionCommandHandler(IApplicationDbContext context, ICbtQuestionParser questionParser)
+        {
+            _context = context;
+
```
Comment (parvez-syn) [inline src/Application/CbtQuestion/Command/Create/CreateCbtQuestionCommand.cs:25]: 💄 remove extra white spaces

File: src/Application/CbtQuestion/Helper/CbtQuestionParser.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+namespace Application.CbtQuestion.Command.Helper
+{
+    public class CbtQuestionParser : ICbtQuestionParser
+    {
+         public List<CbtQuestionVm> ParseQuestions(string csvFilePath)
+        {
+            //csvFilePath = @"E:\Project\cbt-questions.csv";
+            var list = new List<CbtQuestionVm>();
+
+            using (var reader = new StreamReader(csvFilePath))
```
Comment (parvez-syn) [inline src/Application/CbtQuestion/Helper/CbtQuestionParser.cs:15]: 💡 could this be better and in less lines? ```
using (StreamReader sr = File.OpenText(fileName))
{
        string s = String.Empty;
        while ((s = sr.ReadLine()) != null)
        {
            //we're just testing read speeds
        }
}```

File: src/Application/CbtQuestion/Helper/CbtQuestionParser.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                            OptionD = cols[5],
+                            Answer = char.Parse(cols[6])
+
+                        });
+                    }
+                }
+
+            }
+
+            DeleteFile(csvFilePath);
```
Comment (parvez-syn) [inline src/Application/CbtQuestion/Helper/CbtQuestionParser.cs:42]: 👎 Delete file Must be in finally block

File: src/Application/CbtQuestion/Model/CbtQuestionVm.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+namespace Application.CSVReader.Command.Model
+{
+    public class CbtQuestionVm
```
Comment (parvez-syn) [inline src/Application/CbtQuestion/Model/CbtQuestionVm.cs:3]: 🥇

File: src/Application/DependencyInjection.cs [hunk 7]
```diff
@@ -22,7 +24,7 @@ public static IServiceCollection AddApplication(this IServiceCollection services
             services.AddTransient<ICourseService, CourseService>();
 			services.AddTransient<IEmailService, EmailService>();
 			services.AddTransient<ITemplateProcessor, TemplateProcessor>();
-
```
Comment (parvez-syn) [inline src/Application/DependencyInjection.cs:25]: 💡 please format code in this file (CTRL+ALT+F in vscode)

File: src/Application/FileUploads/Commands/Create/CreateFileUploadCommand.cs [hunk 8]
```diff
@@ -28,23 +28,32 @@ public CreateFileUploadCommandHandler(IApplicationDbContext context, IDateTime d
...
+                if(request.IsFileUploaded)
+                {
+                    await _context.FileUpload.AddAsync(entity);
+                    await _context.SaveChangesAsync(cancellationToken);
+                    return entity.FileUploadId;
+                }
                 return entity.FileUploadId;
             }
-            return entity.FileUploadId;
+            catch (System.Exception ex)
```
Comment (parvez-syn) [inline src/Application/FileUploads/Commands/Create/CreateFileUploadCommand.cs:51]: ❓ what 's the need of try-catch if you are just re-throwing exception

File: src/Application/FileUploads/Commands/Create/CreateFileUploadCommand.cs [hunk 9]
```diff
@@ -28,23 +28,32 @@ public CreateFileUploadCommandHandler(IApplicationDbContext context, IDateTime d
...
+                {
+                    FileOriginalName = request.FileOriginalName,
+                    FileNewName = request.FileNewName,
+                    FileServerPath = request.FileServerPath,
+                    FileType = request.FileType,
+                    ActiveFlag = request.ActiveFlag,
+                    CreatedBy = request.CreatedBy,
+                    CreatedOn = _dateTime.Now
+                };
+                if(request.IsFileUploaded)
```
Comment (parvez-syn) [inline src/Application/FileUploads/Commands/Create/CreateFileUploadCommand.cs:43]: ❓ what does `request.IsFileUploaded` mean here

File: src/Domain/Entities/FileUpload.cs [hunk 10]
```diff
@@ -13,5 +11,6 @@ public partial class FileUpload : AuditableEntityWithActiveFlag
         public string FileType { get; set; }
 
         public virtual CourseFileUpload CourseFileUpload { get; set; }
+        public CbtTrainingMetrial CbtSlides { get; set; }
```
Comment (parvez-syn) [inline src/Domain/Entities/FileUpload.cs:14]: ❗ this should be virtual

File: src/WebUI/Controllers/CbtController.cs [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        //TODO To be moved to config
+        private static string FileUploadPath => Path.Combine(Directory.GetCurrentDirectory(), "Upload\\Cbt");
+
+        
+        
+        [HttpPost("Question/{catalogId}")]        
+        public async Task<int> ImportQuestion(IFormFile file, int catalogId, CancellationToken cancellationToken)
+        {
+            string fileNameToUpload = FileNameToSave(file.FileName);
```
Comment (parvez-syn) [inline src/WebUI/Controllers/CbtController.cs:24]: ❗ Move all the logic to either command or service

File: src/WebUI/Controllers/CbtController.cs [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using TALMS.WebUI.Services;
+
+namespace TALMS.WebUI.Controllers
+{
+    public class CbtController : ApiController
+    {
+        //TODO To be moved to config
+        private static string FileUploadPath => Path.Combine(Directory.GetCurrentDirectory(), "Upload\\Cbt");
```
Comment (parvez-syn) [inline src/WebUI/Controllers/CbtController.cs:17]: ❗ Also move FileUploadPath as readonly construstor assigned member to either service or command

**Review by parvez-syn — CHANGES_REQUESTED** [review]
I have come comments, please have a look, let me know if there are questions

File: src/Application/FileUploads/Commands/Create/CreateFileUploadCommand.cs (hunk 9 above)
Comment (SarojKumar8101) [reply src/Application/FileUploads/Commands/Create/CreateFileUploadCommand.cs:43]: Here I am going to check whether file are uploaded or not. if no then ignore to save in db.

File: src/Domain/Entities/FileUpload.cs (hunk 10 above)
Comment (SarojKumar8101) [reply src/Domain/Entities/FileUpload.cs:14]: this is not my code - cbslides

**Pull request closed without merging**